managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2578,6 +2578,7 @@ boolean isOffloadedNeedsDelete(OffloadContext offload, Optional<OffloadPolicies>
         long elapsedMs = clock.millis() - offload.getTimestamp();
         return offloadPolicies.filter(policies -> offload.getComplete() && !offload.getBookkeeperDeleted()
                 && policies.getManagedLedgerOffloadDeletionLagInMillis() != null
+                && policies.getManagedLedgerOffloadDeletionLagInMillis() != -1
```

Comment:
Maybe `policies.policies.getManagedLedgerOffloadDeletionLagInMillis() > -1` is better?

Reply:
updated